helpers/klusterletconfig/merge.go

```diff
@@ -17,6 +17,7 @@ var klusterletConfigMergeFuncs map[string]func(base, override interface{}) (inte
 	"AppliedManifestWorkEvictionGracePeriod": override,
 	"InstallMode":                            override,
 	"BootstrapKubeConfigs":                   override,
+	"HubKubeAPIServerConfig":                 override,
```

Comment:
@elgnay should we use the new override the old for this field?

Reply:
No. the merge strategy is a bit complicated. Let's discuss it offline.